Captain Megaphone

Flaborized opened the pull request:
<!-- The text between the arrows are comments - they won't be visible on your PR. -->
<!-- To label this PR, add the label(s) without the prefixes surrounded by brackets anywhere, for example: [LABEL] -->
<!-- PRs should at least have one area (A-) label and at least one category (C-) label -->

[FEAT]

## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->

Adds a megaphone to the captains locker. Holding the megaphone in your hand and speaking will make your maptext larger and your voice louder! (Larger in chatbox). Since you're the most important person on the station you can now have the easiest to read text to match! (Or have it be stolen and be shamed as it's used against you) 
Has an emag effect 👁️ (Non-gameplay-altering)

Compatible with the red and blue captain medals (if you hold it in your hand while activating)



![](https://cdn.discordapp.com/attachments/1041726307584704532/1101111132187537519/image.png)



![](https://cdn.discordapp.com/attachments/1041726307584704532/1100719080387924018/image.png)



![](https://cdn.discordapp.com/attachments/1041726307584704532/1100720783682187264/image.png)

## Why's this needed? <!-- Describe why you think this should be added to the game. -->

I think it's a fun cute lil feature! Wanted to make this for a while and have finally gotten around to it. Helps the captain feel like they have a "commanding presence". Really just for fun roleplaying purposes though. Can be var edited to make the text much larger if an admin pleases.

Only one on the station so hopefully not a very annoying item either! I tried to balance it being noticeable but not obnoxious.


## Changelog <!-- If necessary, put your changelog entry below. Otherwise, /please/ delete this entire section. -->
<!-- Put how you want to be credited in the changelog in place of CodeDude. -->
<!-- Use (*) for major changes and (+) for minor changes. See the contributor guide for details. For example: -->

```changelog
(u)Flaborized
(+)The captain's locker has been stocked with a megaphone to help them get their finer points across.
```

DisturbHerb commented:
At one point many years ago, one of the HoSii did suggest adding such a megaphone to the Armory. Not sure if that'd fit, but the option is there. 👀

muzcl commented:
Having the emag effect cause ear damage would be nice, not many things deal ear damage in this game I can think of. If its too overpowered like that, you could make it break after a few uses to balance it.

Azrun commented on code/procs/mobprocs/chatprocs.dm:
```diff
@@ -466,6 +466,13 @@
 
+	if (src.find_type_in_hand(/obj/item/megaphone))
+		var/obj/item/megaphone/megaphone = src.find_type_in_hand(/obj/item/megaphone)
+		if	(megaphone.makes_you_quieter)
```
```suggestion
		if (megaphone.makes_you_quieter)
```

Azrun commented on code/mob/living.dm:
```diff
@@ -1104,10 +1104,16 @@
+
+		if (src.find_type_in_hand(/obj/item/megaphone))
+			var/obj/item/megaphone/megaphone = src.find_type_in_hand(/obj/item/megaphone)
+			popup_style = "font-size: [megaphone.maptext_size]px;"
+
```
Should this be `+=` otherwise it will replace the bioeffects?
/datum/bioEffect/speech/comic

Flaborized commented:
Undrafting this since I added the things I think I can add to this. Made the text bold and green (with a slight dark green outline) to better help it stand out and make it look less... thin. Added blue and red variants that you can get by holding it and activating the correct medals. 

As for the sound, I tried for a while to mix something good in audacity but I can't really make something passable! I'm ok adding it without a sound and I can't *personally* provide one I'm happy with in the time frame I wanna work on this for :sheltersweat:

new look:

![](https://cdn.discordapp.com/attachments/1041726307584704532/1101111132187537519/image.png)
